src/main/java/pedido/Cardapio.java

```diff
@@ -5,38 +5,56 @@
 import java.util.TreeMap;
 
 public class Cardapio {
-    private TreeMap<Ingrediente,Double> precos;
+    private TreeMap<Ingrediente, Double> precos;
 
-    public Cardapio(){
-        this.precos= new TreeMap<>();
+    public Cardapio() {
+        this.precos = new TreeMap<>();
     }
 
-    public TreeMap<Ingrediente, Double> getPrecos(){
+    public TreeMap<Ingrediente, Double> getPrecos() {
+
         return this.precos;
     }
 
-    public void adicionarIngrediente(Ingrediente ingrediente,Double preco){
-        //TODO
+    public void adicionarIngrediente(Ingrediente ingrediente, Double preco) {
+
+        if (preco <= 0) {
+            throw new IllegalArgumentException("Preco invalido.");
+        }
+        this.precos.put(ingrediente, preco);
     }
 
-    public boolean atualizarIngrediente(Ingrediente ingrediente,Double preco){
-       //TODO
-        return true;
+
+    public void atualizarIngrediente(Ingrediente ingrediente, Double preco) {
+        if (!precos.containsKey(ingrediente)) {
```

Comment:
para o containsKey tb poderia ter um método específico com tratamento de exceção específica chamando-o dentro dos métodos atualizar, remover, buscar.

Se for feita tratamento de exceção específica, deve-se alterar na classe de teste pq lá as exceções estão generalizadas.